pynemo/nemo_bdy_extr_tm3.py

```diff
@@ -394,21 +409,29 @@ def __init__(self, setup, SourceCoord, DstCoord, Grid, var_nam, grd, pair):
         self.sc_ind = sc_ind
         self.dst_dep = dst_dep
         self.num_bdy = num_bdy
-        self.id_121 = id_121
-        if not self.isslab:
+        self.id_121_2d = id_121_2d
+        self.id_121_3d = id_121_3d
+        if not isslab:
             self.bdy_z = DC.depths[self.g_type]['bdy_H']
         else:
             self.bdy_z = np.zeros([1])
             
         self.dst_z = dst_dep
         self.sc_z_len = sc_z_len
         self.sc_time = sc_time
-        self.tmp_filt = tmp_filt
+        self.tmp_filt_2d = tmp_filt_2d
+        self.tmp_filt_3d = tmp_filt_3d
         self.dist_tot = dist_tot
 
         self.d_bdy = {}
+        
+        # Need to qualify for key_vec
         for v in range(self.nvar):
-            self.d_bdy[self.var_nam[v]] = {}
+            if grd == 'v':
+                 self.d_bdy[self.var_nam[v+1]] = {}
+            else:
+                 self.d_bdy[self.var_nam[v]] = {}
```

Comment:
for some reason in Tom's code either u or v is extracted - we need to keep both for the rotation onto the destination grid. u and v are always index 0 and 1 in Extract object - again a very quick and dirty fix. In future we should have a general Extract Class and use _inheritance_ for the different permutations.